File: src/main/java/com/extollit/gaming/ai/path/model/FlagSampler.java
package com.extollit.gaming.ai.path.model;

public class FlagSampler {
    private final IOcclusionProvider occlusionProvider;
    private int volatileCount;

    public FlagSampler(IOcclusionProvider op) {
        this.occlusionProvider = op;
    }

    public byte flagsAt(int x, int y, int z) {
        byte flags = this.occlusionProvider.elementAt(x, y, z);
        if (volatileIn(flags))
            this.volatileCount++;
        return flags;
    }

    private boolean volatileIn(byte flags) {
        return Logic.doorway.in(flags);
    }

    public int volatility() { return this.volatileCount; }
}

File: src/main/java/com/extollit/gaming/ai/path/model/FlagSampler.kt
package com.extollit.gaming.ai.path.model

class FlagSampler(private val occlusionProvider: IOcclusionProvider?) {
    var volatility = 0
        private set

    fun flagsAt(x: Int, y: Int, z: Int): Byte {
        val flags = occlusionProvider!!.elementAt(x, y, z)
        if (volatileIn(flags)) volatility++
        return flags
    }

    private fun volatileIn(flags: Byte): Boolean = Logic.doorway.flagsIn(flags)
}